libindi/drivers/focuser/sestosenso.cpp

```diff
@@ -88,10 +132,22 @@ bool SestoSenso::initProperties()
     IUFillNumber(&TemperatureN[0], "TEMPERATURE", "Celsius", "%6.2f", -50, 70., 0., 0.);
     IUFillNumberVector(&TemperatureNP, TemperatureN, 1, getDeviceName(), "FOCUS_TEMPERATURE", "Temperature", MAIN_CONTROL_TAB, IP_RO, 0, IPS_IDLE);
 
-//    // Limits
-//    IUFillNumber(&LimitsN[SS_MIN_LIMIT], "FOCUS_MIN_STEP", "Min", "%.f", 0, 10000., 0., 0.);
-//    IUFillNumber(&LimitsN[SS_MAX_LIMIT], "FOCUS_MAX_STEP", "Max", "%.f", 10000,  2097152., 0., 200000.);
-//    IUFillNumberVector(&LimitsNP, LimitsN, 2, getDeviceName(), "FOCUS_LIMITS", "Limits", MAIN_CONTROL_TAB, IP_RW, 0, IPS_IDLE);
+    // Focuser calibration
+    IUFillText(&CalibrationMessageT[0], "CALIBRATION", "Calibration stage", "");
+    IUFillTextVector(&CalibrationMessageTP, CalibrationMessageT, 1, getDeviceName(), "CALIBRATION_MESSAGE", "Calibration", MAIN_CONTROL_TAB, IP_RO, 0, IPS_IDLE);
+
+    IUFillSwitch(&CalibrationS[0],"CALIBRATION_START","Start", ISS_ON);
+    IUFillSwitch(&CalibrationS[1],"CALIBRATION_NEXT","Next", ISS_OFF);
+    IUFillSwitchVector(&CalibrationSP, CalibrationS, 2, getDeviceName(), "FOCUS_CALIBRATION", "Calibration", MAIN_CONTROL_TAB, IP_RW, ISR_1OFMANY, 0, IPS_IDLE);
```

Comment:
Maybe it would make more sense to have ISR_ATMOST1 switch with both switches in the OFF position.

Then we CALIBRATION_START is ON, you begin the calibration process. If it is pressed again, the calibration is cancelled/stopping. 

When CALIBRATION_NEXT is ON, you go to the next step (but it is immediately switched to OFF in the driver so that it can be pressed again in the next step by the user).

Sorry if it is not clear!

Reply:
Understanding exactly how the switches work has been a struggle
I don't actually want to be able to cancel the calibration process. It leaves the focuser in an undefined state. You really have to complete the process once you start.
So if I understand correctly all I need to do is change line 140 to ISS_ON?

Reply:
You can commit to your branch and it will show up here. No need to cancel the PR.

So making them both ISS_ON will not work. They need both to be ISS_OFF and the rule is ISR_ATMOST1 which means that all switches can be off, but only one can be ON at a time.

After user click START, start the process and then set the switch to ON. START=ON, NEXT=OFF

When user clicks NEXT, go to next stage, but keep START=ON and NEXT=OFF

when process is completed, both switches are OFF.

Reply:
Got it. Makes perfect sense. I have set them to OFF, when I start the Calibration process I set START ON and at the end I switch NEXT OFF. I also changed the FastMove to be all OFF.